core/mm/core_mmu.c

```diff
@@ -1346,6 +1432,10 @@ void __weak core_init_mmu_map(unsigned long seed, struct core_mmu_config *cfg)
 	struct tee_mmap_region *tmp_mmap = get_tmp_mmap();
 	unsigned long offs = 0;
 
+	if (IS_ENABLED(CFG_CORE_RELOCATABLE) &&
+	    (core_mmu_tee_load_pa & SMALL_PAGE_MASK))
```

Comment:
Do we have to panic here, or can we just load op-tee on the rounded up page aligned address

Reply:
Do you mean to memmove the entire OP-TEE binary? If so then we should do that before we call relocate() for the first time in `core/arch/arm/kernel/entry_a64.S`. In fact, we should do it before moving `struct boot_embdata`. Do you think it's worth the trouble?